rs/execution_environment/src/execution_environment.rs

```diff
@@ -2226,6 +2235,27 @@ impl ExecutionEnvironment {
         Ok(Encode!(&result).unwrap())
     }
 
+    fn subnet_metrics(
+        &self,
+        replica_version: &ReplicaVersion,
+        args: SubnetMetricsArgs,
+    ) -> Result<Vec<u8>, UserError> {
+        // TODO: Check taken from node_metric_history; but is this actually needed? Can such a call be routed wrong?
```

Comment:
These are both methods that are only available via inter-canister calls, right? In that case, it seems to me like a defensive programming artifact -- you could only ever get into this state if there's another bug at the place where the call is routed in the system api. We can leave it as it doesn't hurt but we might want to treat it somewhat differently (e.g. raise a critical error so that we know if it happens). In any case, I'd keep it consistent with `node_metric_history` and then maybe file a ticket to look into making this a bit saner.